scripts/validate_snapshots.py

```diff
@@ -0,0 +1,110 @@
+"""
+Command line utility to validate snapshots prior to indexing by Azul
+"""
+
+import argparse
+from collections import (
+    defaultdict,
+)
+import json
+import logging
+import sys
+
+from azul import (
+    config,
+)
+from azul.args import (
+    AzulArgumentHelpFormatter,
+)
+from azul.azulclient import (
+    AzulClient,
+)
+from azul.bigquery import (
+    backtick,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+from azul.terra import (
+    TDRSourceSpec,
+)
+
+log = logging.getLogger(__name__)
+configure_script_logging(log)
+
+
+def main(args):
+    joiner = '||'
+    deny_snapshots = defaultdict(dict)
+
+    azul = AzulClient(num_workers=1)
+    sources_by_catalog = {
+        catalog: azul.catalog_sources(catalog)
+        for catalog in args.catalogs
+    }
+    for catalog, sources in sources_by_catalog.items():
+        tdr_plugin = azul.repository_plugin(catalog)
+        log.info('Checking for %r in catalog %s', joiner, catalog)
+        if args.snapshot is not None:
+            sources = [s for s in sources if args.snapshot in s]
+            if len(sources) < 1:
+                continue
+        for spec in sources:
+            log.info('Validating snapshot %s', spec)
+            source = TDRSourceSpec.parse(spec)
+            tables = tdr_plugin._full_table_name(source, 'INFORMATION_SCHEMA.COLUMNS')
+            query = f'SELECT table_name, column_name FROM {backtick(tables)}'
+            rows = tdr_plugin._run_sql(query)
+            table_columns = defaultdict(list)
+            for row in rows:
+                table_columns[row[0]].append(row[1])
+            for table_name, columns in table_columns.items():
```

Comment:
```suggestion
            for table_name, columns in table_columns.items():
                log.info('Validating table %s', table_name)
```
I think the extra log is needed here. With `AZUL_DEBUG=0` there is a long pause between the `Validating snapshot …` log entries, making one unsure if the script has frozen.

Reply:
I intended it that way. The extra detail only slows the overall process which gets worse with the increasing number of snapshots that need to be checked. Even just on the latest lungmap catalog, which isn't one of our biggest catalogs. Plus, the volume of logging when AZUL_DEBUG=1 or 2, would be, IMO, excessive and somewhat duplicative, which is too much. Do you disagree?